Do not grab the synaptics device if not running on the current VT.
This helps when switching user or more in general when running more than one
instance of Xorg.

Index: xserver-xorg-input-synaptics/src/synaptics.c
===================================================================
--- xserver-xorg-input-synaptics.orig/src/synaptics.c	2009-06-03 21:46:59.820825153 +0900
+++ xserver-xorg-input-synaptics/src/synaptics.c	2009-06-06 13:35:52.016859832 +0900
@@ -714,6 +714,9 @@
 
     DBG(3, ErrorF("Synaptics DeviceOn called\n"));
 
+    if (xf86Screens[0]->vtSema == FALSE)
+	    return !Success;
+
     SetDeviceAndProtocol(local);
     local->fd = xf86OpenSerial(local->options);
     if (local->fd == -1) {
